concent_api/core/tests/test_payments_sci_callback.py

```diff
@@ -0,0 +1,192 @@
+                    ):
+                        signed_transaction = sci_callback(transaction)
+
+        self.assertEqual(signed_transaction.v, self.v)
+        self.assertEqual(signed_transaction.r, self.r)
+        self.assertEqual(signed_transaction.s, self.s)
+
+    def test_that_sci_callback_should_raise_exception_on_timeout(self):
+        transaction = self._create_unsigned_transaction()
+
+        with mock.patch('core.payments.sci_callback.socket.socket.connect'):
+            with mock.patch('core.payments.sci_callback.send_over_stream', side_effect=socket.timeout):
+                with self.assertRaises(SCICallbackTimeoutError):
+                    sci_callback(transaction)
+
+    def test_that_sci_callback_should_raise_exception_on_receiving_incorrectly_signed_middleman_protocol_message(self):
+        transaction = self._create_unsigned_transaction()
+
+        with mock.patch('core.payments.sci_callback.socket.socket.connect'):
+            with mock.patch('core.payments.sci_callback.send_over_stream'):
+                with mock.patch('core.payments.sci_callback.unescape_stream'):
+                    with mock.patch('middleman_protocol.message.AbstractFrame.deserialize', side_effect=MiddlemanProtocolError):
+                        with self.assertRaises(SCICallbackFrameError):
+                            sci_callback(transaction)
+
+    def test_that_sci_callback_should_raise_exception_on_receiving_incorrectly_signed_golem_message(self):
+        transaction = self._create_unsigned_transaction()
+
+        with mock.patch('core.payments.sci_callback.socket.socket.connect'):
+            with mock.patch('core.payments.sci_callback.send_over_stream'):
+                with mock.patch('core.payments.sci_callback.unescape_stream'):
+                    with mock.patch('middleman_protocol.message.AbstractFrame.deserialize', side_effect=MessageError):
+                        with self.assertRaises(SCICallbackPayloadError):
+                            sci_callback(transaction)
+
+    def test_that_sci_callback_should_raise_exception_when_messages_request_ids_do_not_match(self):
+        transaction = self._create_unsigned_transaction()
+
+        with mock.patch('core.payments.sci_callback.socket.socket.connect'):
+            with mock.patch('core.payments.sci_callback.send_over_stream'):
+                with mock.patch('core.payments.sci_callback.unescape_stream'):
+                    with mock.patch(
+                        'middleman_protocol.message.AbstractFrame.deserialize',
+                        return_value=GolemMessageFrame(payload=self._create_signed_transaction(), request_id=100),
+                    ):
+                        with self.assertRaises(SCICallbackRequestIdError):
+                            sci_callback(transaction)
+
+    def test_that_sci_callback_should_raise_exception_when_response_is_transaction_rejected_or_error_frame(self):
```

Comment:
In this test `ErrorFrame` case is being checked (please change the name)

Reply:
Updated. :+1: